pkg/netlink/netlink.go

```diff
@@ -194,3 +202,64 @@ func GetDefaultGatewayInterface() (*net.Interface, error) {
 
 	return nil, fmt.Errorf("unable to find default route")
 }
+
+func (n *netlinkType) DeleteIfaceAndAssociatedRoutes(iface string, tableID int) error {
```

Comment:
Since this function uses existing interface methods, I suggest we put it in the package but not part of the interface so it doesn't have to be specifically implemented by the fake.
 
```suggestion
func DeleteIfaceAndAssociatedRoutes(iface string, tableID int) error {
    n := New()
```